File: Array/pyramid.py
n = int(input("Enter a number"))
dp = n+1
li = [[0]*dp for i in range(dp)]
for i in range(0,n):
    for j in range(0,i+1):
        if(j==0):
            li[i][j] = i+1
        elif j%2 != 0:
            li[i][j] = li[i][j-1] + ((n-i)*2 - 1)
        elif j%2 == 0:
            li[i][j] = 1 + max([li[i-1][j],li[i-1][j+1]])
print(li)

File: Array/pyramid.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    int n = 6;
    vector<vector<int>> a(n,vector<int>(n,0));
    for(int i=0;i<n;i++){
        for(int j=0;j<=i;j++){
            if(j==0){
                a[i][j] = i+1;
            }
            else if(j%2 != 0 ){
                a[i][j] = a[i][j-1] + ((n-i)*2 - 1);
            }
            else if(j%2 == 0){
                a[i][j] = 1 + max (a[i-1][j-1],a[i-1][j]);
            }
        }
    }
    for(auto i:a){
        for (int j:i){
            if(j != 0 ){
                cout<<j<<"  ";
            }
        }
        cout<<endl;
    }
}